Firestore/src/Query.php

```diff
@@ -895,11 +946,14 @@ private function newQuery(array $additionalConfig, $overrideTopLevelKeys = false
      * Some optimizations cannot be performed ahead of time and must be done
      * at execution.
      *
-     * @param array $query The incoming query
+     * @internal Only supposed to be used internally.
+     *
+     * @access private
      * @return array The final query data
      */
-    private function finalQueryPrepare(array $query)
+    public function finalQueryPrepare()
```

Comment:
I'd really like to to not do this. Let's look for alternatives. Off the top of my head, we have several options here:
1. PHPUnit lets you mock private and protected methods using `setMethods` on the `MockBuilder` object. Prophesy may do something similar
2. We can make this method `protected` and then subclass it in our tests
3. We can use reflection to access private/protected methods in tests

Reply:
@bshaffer I've moved this function to a new QueryTrait and I kept it private. I had to do some refactoring to get this working.